connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala

```diff
@@ -79,4 +87,79 @@ class KustoClient(val clusterAlias: String, val engineClient: Client, val dmClie
+    import coordinates._
+
+    val mergeTask = Future {
+      try {
+        partitionsResults.value.asScala.foreach {
+          partitionResult =>
+            KDSU.runSequentially[IngestionStatus](
+              func = () => partitionResult.ingestionResult.getIngestionStatusCollection.get(0),
+              0,
+              delayPeriodBetweenCalls,
+              (timeout.toMillis / delayPeriodBetweenCalls + 5).toInt,
+              res => res.status == OperationStatus.Pending,
+              res => res.status match {
+                case OperationStatus.Pending =>
+                  throw new RuntimeException(s"Ingestion to Kusto failed on timeout failure. Cluster: '${coordinates.cluster}', " +
+                    s"database: '${coordinates.database}', table: '$tmpTableName', batch: '$batchIdIfExists', partition: '${partitionResult.partitionId}'")
+                case OperationStatus.Succeeded =>
+                  KDSU.logInfo(myName, s"Ingestion to Kusto succeeded. " +
+                    s"Cluster: '${coordinates.cluster}', " +
+                    s"database: '${coordinates.database}', " +
+                    s"table: '$tmpTableName', batch: '$batchIdIfExists', partition: '${partitionResult.partitionId}'', from: '${res.ingestionSourcePath}', Operation ${res.operationId}")
+                case otherStatus =>
+                  throw new RuntimeException(s"Ingestion to Kusto failed with status '$otherStatus'." +
+                    s" Cluster: '${coordinates.cluster}', database: '${coordinates.database}', " +
+                    s"table: '$tmpTableName', batch: '$batchIdIfExists', partition: '${partitionResult.partitionId}''. Ingestion info: '${readIngestionResult(res)}'")
+              }).await(timeout.toMillis, TimeUnit.MILLISECONDS)
+        }
+
+        // Move data to real table
+        // Protect tmp table from merge/rebuild and move data to the table requested by customer. This operation is atomic.
+        kustoAdminClient.execute(database, generateTableAlterMergePolicyCommand(tmpTableName, allowMerge = false, allowRebuild = false))
+        kustoAdminClient.execute(database, generateTableMoveExtentsCommand(tmpTableName, table.get))
+        KDSU.logInfo(myName, s"write to Kusto table '${table.get}' finished successfully $batchIdIfExists")
+      } catch {
+        case ex: Exception =>
+          KDSU.reportExceptionAndThrow(
+            myName,
+            ex,
+            "Trying to poll on pending ingestions", coordinates.cluster, coordinates.database, coordinates.table.getOrElse("Unspecified table name"),
+            isLogDontThrow = true
+          )
+      } finally {
+        tryCleanupIngestionByproducts(database, kustoAdminClient, tmpTableName)
+      }
+    }
+
+    if(!isAsync)
+    {
+      Await.result(mergeTask, timeout)
+    }
+  }
+
+  private[kusto] def tryCleanupIngestionByproducts(database: String, kustoAdminClient: Client, tmpTableName: String): Unit = {
```

Comment:
In my experience, try usually means it can fail. would consider renaming
https://stackoverflow.com/q/27459937/929164

Reply:
your experience is right - this can fail - and what we mean here that we dont care if it failed or not

Reply:
if you don't care about if it can fail, then remove the try in the name. 
when you read code with `try` prefix, you expect it to be wrapped with try catch..